model/common/src/icon4py/model/common/grid/grid_manager.py

```diff
@@ -26,495 +36,663 @@ def __init__(self, *args, **kwargs):
+    MEAN_DUAL_EDGE_LENGTH = "mean_dual_edge_length"
+    MEAN_CELL_AREA = "mean_cell_area"
+    MEAN_DUAL_CELL_AREA = "mean_dual_cell_area"
+    DOMAIN_LENGTH = "domain_length"
+    DOMAIN_HEIGHT = "domain_height"
+    SPHERE_RADIUS = "sphere_radius"
+    CARTESIAN_CENTER = "domain_cartesian_center"
+
+
+class MandatoryPropertyName(PropertyName):
+    """
+    File attributes present in all files.
+    DWD generated (from [icon-tools](https://gitlab.dkrz.de/dwd-sw/dwd_icon_tools)
+    [grid files](http://icon-downloads.mpimet.mpg.de/dwd_grids.xml) contain only those properties.
+    """
+
+    TITLE = "title"
+    INSTITUTION = "institution"
+    SOURCE = "source"
+    GRID_UUID = "uuidOfHGrid"
+    PARENT_GRID_ID = "uuidOfParHGrid"
+    NUMBER_OF_GRID = "number_of_grid_used"
+    URI = "ICON_grid_file_uri"
+    CENTER = "center"
+    SUB_CENTER = "subcenter"
+    CRS_ID = "crs_id"
+    CRS_NAME = "crs_name"
+    GRID_MAPPING = "grid_mapping_name"
+    ELLIPSOID = "ellipsoid_name"
+    SEMI_MAJOR_AXIS = "semi_major_axis"
+    INVERSE_FLATTENING = "inverse_flattening"
+    LEVEL = "grid_level"
+    ROOT = "grid_root"
+
+
+class DimensionName(GridFileName):
+    """Dimension values (sizes) used in grid file."""
+
+    #: number of vertices
+    VERTEX_NAME = "vertex"
+
+    #: number of edges
+    EDGE_NAME = "edge"
+    #: number of cells
+    CELL_NAME = "cell"
+
+    #: number of edges in a diamond: 4
+    DIAMOND_EDGE_SIZE = "no"
+
+    #: number of edges/cells neighboring one vertex: 6 (for regular, non pentagons)
+    NEIGHBORS_TO_VERTEX_SIZE = "ne"
+
+    #: number of cells edges, vertices and cells neighboring a cell: 3
+    NEIGHBORS_TO_CELL_SIZE = "nv"
 
-        #: name of V2E2V connectivity in gridfile: dims(ne=6, vertex),
-        #: all vertices of a pentagon/hexagon, same as V2C2V
-        V2E2V = "vertices_of_vertex"  # does not exist in simple.py
+    #: number of vertices/cells neighboring an edge: 2
+    NEIGHBORS_TO_EDGE_SIZE = "nc"
 
-        #: name of V2E dimension in grid file: dims(ne=6, vertex)
-        V2E = "edges_of_vertex"
+    #: number of child domains (for nesting)
+    MAX_CHILD_DOMAINS = "max_chdom"
 
-        #: name fo V2C connectivity in grid file: dims(ne=6, vertex)
-        V2C = "cells_of_vertex"
+    #: Grid refinement: maximal number in grid-refinement (refin_ctl) array for each dimension
+    CELL_GRF = "cell_grf"
+    EDGE_GRF = "edge_grf"
+    VERTEX_GRF = "vert_grf"
 
-        #: name of E2V connectivity in grid file: dims(nc=2, edge)
-        E2V = "edge_vertices"
 
-        #: name of C2V connectivity in grid file: dims(nv=3, cell)
-        C2V = "vertex_of_cell"  # does not exist in grid.simple.py
+class FieldName(GridFileName):
+    ...
 
-        #: name of E2C connectivity in grid file: dims(nc=2, edge)
-        E2C = "adjacent_cell_of_edge"
 
-        #: name of C2E connectivity in grid file: dims(nv=3, cell)
-        C2E = "edge_of_cell"
+class ConnectivityName(FieldName):
```

Comment:
Sorry for another late comment. Is it good to have two separate classes such as IndicesFieldName for all connectivities and FiedlName for other float varaibles that indicate coordinates etc?

Reply:
This is more a semantic oriented decision really not technically based on the netcdf structure. For netcdf those are just fields.